Mage/src/main/java/mage/abilities/dynamicvalue/common/GetKickerXValue.java

```diff
@@ -19,35 +17,9 @@ public enum GetKickerXValue implements DynamicValue {
 
     @Override
     public int calculate(Game game, Ability sourceAbility, Effect effect) {
-        // calcs only kicker with X values
-
-        // kicker adds additional costs to spell ability
-        // only one X value per card possible
-        // kicker can be calls multiple times (use getKickedCounter)
-
-        int countX = 0;
-        Spell spell = game.getSpellOrLKIStack(sourceAbility.getSourceId());
-        if (spell != null && spell.getSpellAbility() != null) {
-            int xValue = spell.getSpellAbility().getManaCostsToPay().getX();
-            for (Ability ability : spell.getAbilities()) {
-                if (ability instanceof KickerAbility) {
-
-                    // search that kicker used X value
-                    KickerAbility kickerAbility = (KickerAbility) ability;
-                    boolean haveVarCost = kickerAbility.getKickerCosts()
-                            .stream()
-                            .anyMatch(varCost -> !((OptionalAdditionalCostImpl) varCost).getVariableCosts().isEmpty());
-
-                    if (haveVarCost) {
-                        int kickedCount = ((KickerAbility) ability).getKickedCounter(game, sourceAbility);
-                        if (kickedCount > 0) {
-                            countX += kickedCount * xValue;
-                        }
-                    }
-                }
-            }
-        }
-        return countX;
+        // Currently identical logic to the Manacost X value
+        // which should be fine since you can only have one X at a time
+        return (int)CardUtil.getSourceCostTags(game, sourceAbility).getOrDefault("X",0);
```

Comment:
`getSourceCostTags` always returns a map, it will never be null.
`getOrDefault` will return the default instead of a null.

Thus, this code will never cause null conversion. If an X value was not set during casting (such as by casting a spell without paying its mana cost) then there will be no tag. The value of X must be returned as 0 in that situation, not null.